src/MoBi.Core/Domain/Services/RenameInSimulationTask.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MoBi.Core.Domain.Extensions;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Services;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Domain.Services
+{
+   public interface IRenameInSimulationTask
+   {
+      /// <summary>
+      ///    Find building blocks in all simulations with <paramref name="oldName" /> and rename them to match the
+      ///    <paramref name="templateBuildingBlock" /> name
+      /// </summary>
+      void RenameInSimulationUsingTemplateBuildingBlock(string oldName, IBuildingBlock templateBuildingBlock);
+
+      /// <summary>
+      ///    Find modules in all simulations with <paramref name="oldName" /> and rename them to match the
+      ///    <paramref name="templateModule" /> name
+      /// </summary>
+      void RenameInSimulationUsingTemplateModule(string oldName, Module templateModule);
+   }
+
+   public class RenameInSimulationTask : IRenameInSimulationTask
+   {
+      private readonly IMoBiProjectRetriever _projectRetriever;
+
+      public RenameInSimulationTask(IMoBiProjectRetriever projectRetriever)
+      {
+         _projectRetriever = projectRetriever;
+      }
+
+      public void RenameInSimulationUsingTemplateBuildingBlock(string oldName, IBuildingBlock templateBuildingBlock)
+      {
+         getBuildingBlocksWithMatchingNameAndType(_projectRetriever.Current.Simulations, oldName, templateBuildingBlock).Each(simulationAndBuildingBlock =>
```

Comment:
not sure why you are passing the simulations as parameter to this method since it comes from member . 
Dont make the method static and you are good to go